SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp

```diff
@@ -239,13 +281,21 @@ void USpatialReceiver::OnAddComponent(const Worker_AddComponentOp& Op)
 		return;
 	}
 
+	// Queue owner-only components until we have authority
+	if (ClassInfoManager->GetCategoryByComponentId(Op.data.component_id) == SCHEMA_OwnerOnly &&
+		!StaticComponentView->HasAuthority(Op.entity_id, SpatialConstants::CLIENT_ENDPOINT_COMPONENT_ID))
+	{
+		PendingOwnerOnlyComponents.AddUnique(PendingAddComponentWrapper(Op.entity_id, Op.data.component_id, MakeUnique<DynamicComponent>(Op.data)));
```

Comment:
Can these be added and removed before gaining authority?

Reply:
I have thought and thought some more, and I think we can fix this in a different way, so will try a small rewrite.